article/article.md

```diff
@@ -0,0 +1,333 @@
+        }
+    }
+    generateProtoTasks {
+        all().each { task ->
+            task.plugins {
+                javalite {}
+                grpc {
+                    // Options added to --grpc_out
+                    option 'lite'
+                }
+            }
+        }
+    }
+}
+```
+
+To build the application the Android way, I used Architecture Components and Kotlin couroutines, along with the [ViewModel kotlin extension](https://medium.com/androiddevelopers/easy-coroutines-in-android-viewmodelscope-25bffb605471).
+
+Here is the `getPokemon` call inside the `PokedexViewModel`, launched with the `ViewModelScope` in a coroutine, the response is then passed on to LiveData which is observed by my Activity:
+
+```Kotlin
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import io.grpc.ManagedChannelBuilder
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class PokedexViewModel : ViewModel() {
+
+    val pokemonLiveDate: LiveData<GetPokemonResult> get() = _pokemonLiveData
+    private val _pokemonLiveData = MutableLiveData<GetPokemonResult>()
+
+    fun getPokemon(host: String, portStr: String, message: String) {
+        viewModelScope.launch(Dispatchers.IO) {
+            try {
+                val port = if (portStr.isEmpty()) 0 else portStr.toInt()
+                val channel = ManagedChannelBuilder.forAddress(host, port).usePlaintext().build()
+                val stub = PokedexGrpc.newBlockingStub(channel)
+                val request = PokedexRequest.newBuilder().setEnglishName(message.trim()).build()
+                val reply = stub.getPokemon(request)
+                _pokemonLiveData.postValue(GetPokemonResult.Success(reply))
+                channel.shutdown()
+            } catch (e: Exception) {
+                e.printStackTrace()
+                _pokemonLiveData.postValue(GetPokemonResult.Error(e))
+            }
+        }
+    }
+}
+
+sealed class GetPokemonResult {
+    data class Success(val reply: PokedexReply) : GetPokemonResult()
+    data class Error(val exception: Exception) : GetPokemonResult()
+}
+```
+
+To test this implementation, start the application and connect to the Pokédex server and search the Pokémon!
+
+![pokedex-app](pokedex-app.png)
+
+# Conclusion
+
+Building a mobile application is actually harder than it appears. We are constantly concerning about things such as battery usage, celluar data consumption, connection latency, device storage or security. The list goes on and on. A good mobile application should take account of all these criterias. But our current solutions are full of vulnerabilities. 
+
+ We start to see more and more production applications using gRPC. gRPC is still a little bit premature but it is also quite promising. It gives us a perspective on how to tackle mobile platform's restrictions differently.
```

Comment:
If you take into account the previous suggestion, this sentence can be removed ❌